docs/docs/getting_started/tooling/noir_codegen.md

```diff
@@ -0,0 +1,126 @@
+---
+title: Noir Codegen for Typescript
+description: Learn how to use Noir codegen to generate Typescript bindings 
+keywords: [Nargo, Noir, compile, TypeScript]
+sidebar_position: 2
+---
+
+You can generate TypeScript bindings with `noir_codegen`, allowing for easier and faster integration of your Noir circuits into TypeScript projects.
+
+It is used in conjunction with `nargo export` which generates JSON for specified functions.
+
+## Installation
+
+You can install `noir_codegen` into your project using `yarn` or `npm`:
+
+```bash
+yarn add @noir-lang/noir_codegen
+```
+
+or 
+
+```bash
+npm install @noir-lang/noir_codegen
+```
+## Usage
+
+### Export circuit ABI
+
+Before generating TypeScript bindings, you need to export your Noir programs using `nargo export`:
+
+```bash
+nargo export
```

Comment:
I am running `nargo export` in the dir with my Nargo.toml, using today's nightly. 
![image](https://github.com/noir-lang/noir/assets/18372439/fee66699-6558-4c9d-9ed1-ed7cb798dbd8)


![image](https://github.com/noir-lang/noir/assets/18372439/195f81d7-dbf8-435c-b4c8-0cd351c46f8d)

It doesn't generate any files. Any ideas?

Reply:
Ended up fleshing out the doc. `nargo export` works only for libs with functions marked with `#[export]` macro.
Added details about how .json files are generated and consolidated in a single .ts file using the noir-codegen module.